File: rb/features.rb
# CreditCardValidation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CreditCardValidationFeatures
  def self.make_feature(name)
    case name
    when "base"
      CreditCardValidationBaseFeature.new
    when "test"
      CreditCardValidationTestFeature.new
    else
      CreditCardValidationBaseFeature.new
    end
  end
end

File: py/creditcardvalidation_sdk/features.py
# CreditCardValidation SDK feature factory

from creditcardvalidation_sdk.feature.base_feature import CreditCardValidationBaseFeature
from creditcardvalidation_sdk.feature.test_feature import CreditCardValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CreditCardValidationBaseFeature(),
        "test": lambda: CreditCardValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
